test/module/webpage/load-images-no.js

```
@@ -0,0 +1,52 @@
+//! phantomjs: --load-images=no
+
+// Compiled for debug, this tests for leaks when not loading images.
+//
+
+var url = TEST_HTTP_BASE + 'load-images.html';
+
+function do_test() {
+    var page = require('webpage').create();
+
+    var numImagesRequested = 0;
+    var numImagesLoaded = 0;
+    
+    page.onResourceRequested = this.step_func(function (resource) {
+        if (/\.png$/.test(resource.url)) ++numImagesRequested;
+    });
+
+    page.onResourceReceived = this.step_func(function (resource) {
+        if (resource.stage === 'end' && /\.png$/.test(resource.url)) ++numImagesLoaded;
+    });
+
+    function checkImageCounts ()
+    {
+        var imageCounts = page.evaluate (function() {    
+            "use strict";
+            try
+            {
+                var images = document.getElementsByTagName('img');
+                var imageCounts = {};
+                imageCounts.numImages = images.length;
+                imageCounts.numComplete = Array.prototype.reduce.call(images, function(n, image) { return n + (image.complete? 1 : 0); }, 0);
+                return imageCounts;
+            }
+            catch(err)
+            {
+                return undefined;
```

I'm not sure what happens if an exception is thrown and not caught inside `evaluate` -- but this way, the exception is swallowed and we just have `undefined` in a comparison later, which is safe (`assert_*` use strict equality) but might be baffling.  I guess what I want is to make sure that any potential exception gets logged somehow.